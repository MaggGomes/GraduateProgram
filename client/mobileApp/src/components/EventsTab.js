import React, { Component } from 'react';
import { ScrollView, FlatList, View, Image, Text, StyleSheet,  ActivityIndicator, StatusBar } from 'react-native';
import EventCardList from './EventCardList';
import {connect} from 'react-redux';
import {fetchEvents} from '../actions/eventsActions'

/* Events Tab */
export class EventsTab extends Component {
    componentDidMount(){
        this.props.dispatch(fetchEvents());
    }

    _renderItem = ({item}) => (
        <View>
            <Text style={styles.event}>{item.name}</Text>
            <EventCardList id={item._id} name={item.name} days={item.days} navigation={this.props.navigation} />
        </View>
    );

    _keyExtractor = (item) => item._id;

    render() {
        return (
            <ScrollView style={styles.wrapper}>
                <StatusBar
                    backgroundColor="#01308d"
                    barStyle="light-content"
                />
                <Image
                    source={require('../static/img/banner.png')}
                    style={styles.banner}
                />
                {this.props.events.loading && <View style={styles.indicator}><ActivityIndicator size="large" color="#0145CD" /></View>}
                {!this.props.events.loading && this.props.events.error !== null && <Text>{this.props.events.error.message}</Text>}
                {!this.props.events.loading && this.props.events.error === null &&
                <FlatList
                    data={this.props.events.data}
                    renderItem={this._renderItem}
                    keyExtractor={this._keyExtractor}
                />}
            </ScrollView>
        );
    }
}

const styles = StyleSheet.create({
    banner: {
        height:180,
        width:'100%'
    },
    wrapper: {
        flex: 1,
        marginBottom: 20
    },
    event: {
        color: '#494949',
        fontSize: 15,
        paddingLeft: 20,
        paddingTop: 20,
        paddingBottom: 10
    },
    indicator: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    }
});

export default connect(store => ({events: store.events}))(EventsTab);
