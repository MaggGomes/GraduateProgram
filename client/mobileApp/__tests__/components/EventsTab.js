import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import EventsTab from '../../src/components/mochComponents/EventsTab';

configure({ adapter: new Adapter() });

const middleware = [thunk];
const mockStore = configureStore(middleware);

const initialState = {
    events: {
        data: [],
        loading: false,
        error: null
    }
};

describe('EventsTab  component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <EventsTab />,
            { context: { store: mockStore(initialState) } },
        );

        expect(wrapper.dive()).toMatchSnapshot();
    });
});
