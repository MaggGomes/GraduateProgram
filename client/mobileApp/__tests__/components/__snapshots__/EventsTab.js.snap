// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsTab  component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventsTab
    dispatch={[Function]}
    events={
      Object {
        "data": Array [],
        "error": null,
        "loading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          backgroundColor="#01308d"
          barStyle="light-content"
          showHideTransition="fade"
        />,
        false,
        false,
        <FlatList
          data={Array []}
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
      ],
      "style": Object {
        "flex": 1,
        "marginBottom": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "backgroundColor": "#01308d",
          "barStyle": "light-content",
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            backgroundColor="#01308d"
            barStyle="light-content"
            showHideTransition="fade"
          />,
          false,
          false,
          <FlatList
            data={Array []}
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />,
        ],
        "style": Object {
          "flex": 1,
          "marginBottom": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "backgroundColor": "#01308d",
            "barStyle": "light-content",
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
